src/postgres.c

```diff
@@ -88,65 +79,46 @@ int postgres_init() {
  */
 static void *postgres_worker(void *arg) {
   int id = *(int*)arg;
+  size_t nread = 0;
   PGconn *conn = conns[id];
```

Comment:
Since only the worker uses this now it would be nicer to define a "worker data" struct for the id, pg connection and pipe. Then we can kill the globals.